sklearn/tree/export.py

```diff
@@ -412,6 +416,12 @@ def recurse(tree, node_id, criterion, parent=None, depth=0):
                                "; ".join(r for r in ranks[rank]) + "} ;\n")
         out_file.write("}")
 
+        # If out_file is a StringIO buffer, return its contents
```

Comment:
Ah yes this was a mistake, now it should only return the string represenation if `out_file is None`